src/overlays/actors/ovl_En_Bigokuta/z_en_bigokuta.c

```diff
@@ -28,99 +43,626 @@ const ActorInit En_Bigokuta_InitVars = {
+        this->bodyCollider.dim.pos.z = this->shellCollider.dim.pos.z;
+
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->bodyCollider.base);
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->shellCollider.base);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->shellCollider.base);
+        Actor_SetHeight(&this->actor, 82.5f);
+    }
+
+    if (this->drawEffectAlpha > 0.0f) {
+        if (this->drawEffect != BIGOCTO_DRAWEFFECT_ICE) {
+            Math_StepToF(&this->drawEffectAlpha, 0.0f, 0.05f);
+            this->drawEffectScale = (this->drawEffectAlpha + 1.0f) * 0.6f;
+            this->drawEffectScale = CLAMP_MAX(this->drawEffectScale, 1.2f);
+        } else if (!Math_StepToF(&this->drawEffectSteamScale, 1.2f, 0.030000001f)) {
+            func_800B9010(&this->actor, NA_SE_EV_ICE_FREEZE - SFX_FLAG);
+        }
+    }
+}
+
+s32 EnBigokuta_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                                Actor* thisx, Gfx** gfx) {
+    if (limbIndex == 10) {
+        EnBigokuta* this = THIS;
+        s32 envColor;
+        s16 rotX;
+        f32 lastFrame;
+
+        if (this->actionFunc == EnBigokuta_PlayDeathEffects) {
+            lastFrame = Animation_GetLastFrame(&gBigOctoDeathAnim);
+            envColor = ((255.0f / lastFrame) * (lastFrame - this->skelAnime.curFrame));
+        } else {
+            envColor = 255;
+        }
+        gDPPipeSync((*gfx)++);
+        gDPSetEnvColor((*gfx)++, envColor, envColor, envColor, envColor);
+
+        if (this->actionFunc == EnBigokuta_SuckInPlayer) {
+            rotX = (s16)(this->timer * 6144.0f * (1 / 9.0f));
+            rot->x -= rotX;
+        } else if (this->actionFunc == EnBigokuta_HoldPlayer) {
+            if (this->timer == 1) {
+                rotX = 0;
+            } else if (this->timer == 0) {
+                rotX = -0xC00;
+            } else if (this->timer > 0) {
+                rotX = 0x1800;
+            } else {
+                rotX = ((this->timer + 0x18) * 0.041666668f * -6144.0f);
+            }
+            rot->x -= rotX;
+        }
+
+    } else if (limbIndex == 17) {
+        EnBigokuta* this = THIS;
+
+        if (this->actionFunc == EnBigokuta_PlayDeathEffects) {
+            if (this->timer < 5) {
+                Matrix_Scale(1.0f, 1.0f, (this->timer * 0.2f * 0.25f) + 1.0f, MTXMODE_APPLY);
+            } else if (this->timer < 8) {
+                f32 temp = (this->timer - 5) * (1 / 12.0f);
+
+                Matrix_Scale(1.0f + temp, 1.0f + temp, 1.25f - temp, MTXMODE_APPLY);
+            } else {
+                Matrix_Scale(1.25f - ((this->timer - 8) * 0.125f), 1.25f - ((this->timer - 8) * 0.125f), 1.0f,
+                             MTXMODE_APPLY);
+            }
+        } else if (this->actionFunc == EnBigokuta_SuckInPlayer) {
+            f32 temp = sin_rad(this->timer * (M_PI / 3.0f)) * 0.5f;
```

Comment:
Does it make a difference that all of these temps share a name? if not I think you can name them a bit more distinctly. (E.g. `sin` for this one.)